src/cleanvision/issue_managers/image_property_issue_manager.py

```diff
@@ -158,99 +172,113 @@ def find_issues(
 
-        # update info
-        self.update_info(agg_computations, imagelab_info)
+        agg_computations = update_df(agg_computations, new_computations)
 
-        scores = self.compute_scores()
+        ordered_issue_types = self._get_dependency_sorted(self.issue_types)
 
-        self.update_issues(scores, filepaths)
+        self.update_issues(agg_computations, ordered_issue_types)
+        self.update_info(agg_computations)
         self.update_summary()
         return
 
-    def update_issues(self, scores: Dict[str, pd.Series], filepaths: List[str]) -> None:
-        self.issues = pd.DataFrame(index=filepaths)
-        for issue_type in self.issue_types:
-            self.issues = self.issues.join(
-                scores[issue_type].rename(f"{issue_type}_score")
-            )
-            is_issue = self.image_properties[issue_type].mark_issue(
-                scores[issue_type], self.params[issue_type].get("threshold")
-            )
-            self.issues = self.issues.join(
-                is_issue.rename(get_is_issue_colname(issue_type))  # type: ignore
-            )
+    def _get_dependency_sorted(self, issue_types: List[str]) -> List[str]:
+        # todo: remove this hacky way, add an ordering
+        return sorted(issue_types, reverse=True)
 
-    def compute_scores(self) -> Dict[str, pd.Series]:
-        scores = {}
-        for issue_type in self.issue_types:
-            score_column = self.image_properties[issue_type].score_column
-            if score_column in self.info["statistics"]:
-                values = self.info["statistics"][score_column]
+    def update_issues(
+        self, agg_computations: pd.DataFrame, issue_types: List[str]
+    ) -> None:
+        for issue_type in issue_types:
+            score_column_names = self.image_properties[issue_type].score_columns
+            score_columns = agg_computations[score_column_names]
+
+            # todo: this is hacky
+            if issue_type == IssueType.BLURRY.value:
+                if not {
+                    get_is_issue_colname(IssueType.DARK.value),
+                    get_score_colname(IssueType.DARK.value),
+                }.issubset(self.issues):
+                    dark_issue_scores = self.image_properties[
+                        IssueType.DARK.value
+                    ].get_scores(
+                        agg_computations[
+                            self.image_properties[IssueType.DARK.value].score_columns
+                        ],
+                        IssueType.DARK.value,
+                        **self.params[IssueType.DARK.value],
+                    )
+                    is_dark_issue = self.image_properties[
+                        IssueType.DARK.value
+                    ].mark_issue(
+                        dark_issue_scores,
+                        self.params[IssueType.DARK.value].get("threshold"),
+                        IssueType.DARK.value,
+                    )
+                else:
```

Comment:
could you add a comment here to explain the if/else part, eg. 
"\# dark issue score already computed so we can reuse it in the blurry score which depends on it"